### PR TITLE
Use katcp-codec for encoding and decoding messages

### PR DESCRIPTION
**Pull request by bmerry**
I've made this as a beta release (1.10.0b1) because katcp-codec is still at beta.

I also ditched katversion rather than wait for https://github.com/ska-sa/katversion/pull/31, so that I could make the release.

### CODE REVIEW

**Comment by bmerry** [comment]
@ludwigschwardt I requested your review just for the packaging side of things.

**Review by james-smith-za — APPROVED** [review]

File: src/aiokatcp/core.py [hunk 1]
```diff
@@ -581,39 +579,21 @@ def parse(cls, raw) -> "Message":
...
-            if not match:
-                raise KatcpSyntaxError("could not parse name and message ID")
-            name = match.group(1).decode("ascii")
-            mid_raw = match.group(2)
-            if mid_raw is not None:
-                mid = int(mid_raw)
-            else:
-                mid = None
-            mtype = cls._REVERSE_TYPE_SYMBOLS[clean[:1]]
+                raise ValueError("message does not end with newline")
```
Comment (ludwigschwardt) [inline src/aiokatcp/core.py:571]: Why not `KatcpSyntaxError` anymore?

**Review by ludwigschwardt — APPROVED** [review]
Alas, poor katversion, I knew him...

**Comment by bmerry** [comment]
> Also, out of interest, how significant is the speedup with katcp-codec?

In microbenchmarks (particularly large requests like setting gains) it can be 5x, although in practice a lot of the time is spent elsewhere anyway, particularly converting Python objects (strings, floats) to/from raw bytes.

**Pull request merged**